atef/widgets/config/run_base.py

```diff
@@ -169,6 +167,38 @@ def get_relevant_configs_comps(
     return matched_c
 
 
+def get_prepared_step(
+    prepared_file: PreparedProcedureFile,
+    step: ProcedureStep
+) -> List[PreparedProcedureStep]:
+    """
+    Gather all PreparedProcedureStep dataclasses the correspond to the original
+    ProcedureStep.
+
+    Only relevant for active checkouts.
+
+    Parameters
+    ----------
+    prepared_file : PreparedProcedureFile
+        the PreparedProcedureFile to search through
+    step : Union[ProcedureStep, ProcedureGroup]
+        the step to match
+
+    Returns
+    -------
+    List[Union[PreparedProcedureStep, PreparedProcedureGroup]]
+        the PreparedProcedureSteps related to ``step``
+    """
+    # As of the writing of this docstring, this helper is only expected to return
+    # lists of length 1.  However in order to match the passive checkout workflow,
+    # we still return a list of relevant steps.
+    matched_steps = []
+    for pstep in walk_steps(prepared_file.root):
+        if pstep.origin == step:
```

Comment:
Hmm, do we want an equality check here or an `is` check?

Dataclass equality checks implicitly do the [following](https://docs.python.org/3/library/dataclasses.html):
> eq: If true (the default), an __eq__() method will be generated. This method compares the class as if it were a tuple of its fields, in order. Both instances in the comparison must be of the identical type.

Whereas an `is` check would add the prepared step only if its origin were the exact object.